packages/pigeon/lib/objc_generator.dart

```diff
@@ -931,10 +923,10 @@ static id GetNullableObjectAtIndex(NSArray* array, NSInteger key) {
   void writeDataClassExtension(Class klass) {
     final String className = _className(options.prefix, klass.name);
     indent.writeln('@interface $className ()');
-    indent.writeln('+ ($className *)fromMap:(NSDictionary *)dict;');
+    indent.writeln('+ ($className *)fromList:(NSMutableArray *)list;');
```

Comment:
`NSArray *`

You very rarely want to pass around `NSMutable*` objects; for arguments it's only if the function is going to modify it (e.g., `addStuffToArray:(NSMutableArray*)`), and for return values it's only if there's a really compelling reason related to the intended use (e.g., it's creating the initial elements for a mutable array ivar).